From 1982e0e1e06a710a5f9dc3812fcfe43d50d30100 Mon Sep 17 00:00:00 2001
From: Dongdong Zhang <zhangdongdong@eswincomputing.com>
Date: Thu, 15 May 2025 10:15:20 +0800
Subject: [PATCH 547/652] feat:Reserve memory for version info

Changelogs:

1. Add version_table node in the device tree,
   and reserve a piece of memory for fitimage version
   and firmware version.

Signed-off-by: Dongdong Zhang <zhangdongdong@eswincomputing.com>
Co-authored-by: qinhaijun <qinhaijun@eswincomputing.com>
Reviewed-by: Joey Zheng <zhengyu@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/vpu7702-pcie.dts | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts b/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
index 2e6733f627cf..f9b1608c91d4 100644
--- a/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
+++ b/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
@@ -156,7 +156,12 @@ ipcm_vdev0buffer:vdev0buffer@100412000 {
 		};
 
 		rsc_table0:rsc_table0@104412000 {
-			reg = <0x1 0x4412000 0 0x100>;
+			reg = <0x1 0x4412000 0 0x1000>;
+			no-map;
+		};
+
+		version_table:version_table@104413000 {
+			reg = <0x1 0x4413000 0 0x1000>;
 			no-map;
 		};
 
@@ -164,12 +169,10 @@ share_memory {
 			compatible = "eswin,shm";
 			#address-cells = <2>;
 			#size-cells = <2>;
-
-			phys-base = <0x1 0x4412100>;
-			virt-base = <0x1 0x4412100>;
-			shm-size = <0xbbedf00>;
-			reg = <0x1 0x4412100 0 0xbbedf00>;
-			no-map;
+			phys-base = <0x1 0x4414000>;
+			virt-base = <0x1 0x4414000>;
+			shm-size = <0xbbea000>;
+			reg = <0x1 0x4414000 0 0xbbea000>;
 		};
 
 	};
-- 
2.52.0

